contracts/Enigma.sol

```diff
@@ -170,16 +210,32 @@ contract Enigma is EnigmaStorage, EnigmaEvents, Getters, Ownable {
     * worker. This should be called by every worker (and the principal)
     * node in order to receive tasks.
     *
+    * @param _stakingAddress The operating address
     * @param _signer The signer address, derived from the enclave public key
     * @param _report The RLP encoded report returned by the IAS
     * @param _signature Signature
     */
-    function register(address _signer, bytes memory _report, bytes memory _signature)
+    function register(address _stakingAddress, address _signer, bytes memory _report, bytes memory _signature)
     public
     isUpdatedEnigmaContract
+    workerUnregistered(msg.sender)
     isUniqueSigningKey(_signer)
+    areOperatingAndStakingDiff(_stakingAddress)
     {
-        WorkersImpl.registerImpl(state, _signer, _report, _signature);
+        WorkersImpl.registerImpl(state, _stakingAddress, _signer, _report, _signature);
+    }
+
+    /**
+    * Unregisters a staking address' worker.
+    *
+    */
+    function unregister()
```

Comment:
Please enable that both the operating address and the staking address can unregister a worker. Possible use case for that: a worker mistyped the staking address..

Reply:
good point, can add this and push an update!

Reply:
Thanks, this will be great